engine/src/main/java/org/hibernate/search/engine/backend/document/spi/UserIndexFieldConverter.java

```diff
@@ -56,24 +56,51 @@ public Object convertFromProjection(F projection, FromIndexFieldValueConvertCont
 	}
 
 	/**
-	 * Determine whether another converter is DSL-compatible with this one,
-	 * i.e. its {@link #convertFromDsl(Object, ToIndexFieldValueConvertContext)} method is guaranteed
-	 * to always return the same value as this converter's when given the same input.
+	 * Determine whether another converter's {@link #convertFromDsl(Object, ToIndexFieldValueConvertContext)}
+	 * method is compatible with this one's,
+	 * i.e. the method is guaranteed to always return the same value as this converter's when given the same input.
 	 * <p>
-	 * Note: this method is separate from {@link #equals(Object)} because it might return true for two different objects,
-	 * e.g. two objects that implement {@link #convertFromProjection(Object, FromIndexFieldValueConvertContext)} differently.
+	 * Note: this method is separate from {@link #equals(Object)} because it might return {@code true} for two different objects,
+	 * e.g. two objects that implement {@link #convertFromProjection(Object, FromIndexFieldValueConvertContext)} differently,
+	 * but still implement {@link #convertFromDsl(Object, ToIndexFieldValueConvertContext)} in a compatible way.
 	 *
 	 * @param other Another {@link UserIndexFieldConverter}.
-	 * @return {@code true} if the given converter is DSL-compatible.
+	 * @return {@code true} if the given converter's
+	 * {@link #convertFromDsl(Object, ToIndexFieldValueConvertContext)} method is compatible.
 	 * {@code false} otherwise, or when in doubt.
 	 */
-	public boolean isDslCompatibleWith(UserIndexFieldConverter<?> other) {
+	public boolean isConvertFromDslCompatibleWith(UserIndexFieldConverter<?> other) {
 		if ( other == null ) {
 			return false;
 		}
 		return dslToIndexConverter.isCompatibleWith( other.dslToIndexConverter );
 	}
 
+	/**
+	 * Determine whether another converter's {@link #convertFromProjection(Object, FromIndexFieldValueConvertContext)}
+	 * method is compatible with this one's,
+	 * i.e. the method is guaranteed to always return the same value as this converter's when given the same input.
+	 * <p>
+	 * Note: this method is separate from {@link #equals(Object)} because it might return {@code true} for two different objects,
+	 * e.g. two objects that implement {@link #convertFromDsl(Object, ToIndexFieldValueConvertContext)} differently,
+	 * but still implement {@link #convertFromProjection(Object, FromIndexFieldValueConvertContext)} in a compatible way.
+	 *
+	 * @param other Another {@link UserIndexFieldConverter}.
+	 * @return {@code true} if the given converter's
+	 * {@link #convertFromProjection(Object, FromIndexFieldValueConvertContext)} method is DSL-compatible.
+	 * {@code false} otherwise, or when in doubt.
+	 */
+	public boolean isConvertFromProjectionCompatibleWith(UserIndexFieldConverter<?> other) {
```

Comment:
Is it really `from` projection or `to` projection?

Maybe `isDslToIndexConversionCompatibleWith` and `isIndexToProjectionConversionCompatibleWith` would be more clear?

Reply:
It would, but then I'd have to rename the other methods to be consistent.

Please confirm this works for you before I rename everything:

* `convertDslToIndex`
* `isConvertDslToIndexCompatibleWith`
* `convertIndexToProjection`
* `isConvertIndexToProjectionCompatibleWith`

Reply:
Makes sense to me. It's a bit longer but it's definitely more clear about what it's supposed to do.

Reply:
Added a commit to fix this. Waiting for CI to go green, then I'll merge.